src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -207,92 +207,110 @@ const connectInfiniteHits: InfiniteHitsConnector = function connectInfiniteHits(
     return {
       $$type: 'ais.infiniteHits',
 
-      init({ instantSearchInstance, helper }) {
-        const cachedHits = cache.read({ state: helper.state }) || {};
-        showPrevious = getShowPrevious(helper, cachedHits);
-        showMore = getShowMore(helper, cachedHits);
-        sendEvent = createSendEventForHits({
-          instantSearchInstance,
-          index: helper.getIndex(),
-          widgetType: this.$$type!,
-        });
-        bindEvent = createBindEventForHits({
-          index: helper.getIndex(),
-          widgetType: this.$$type!,
-        });
-
+      init(initOptions) {
         renderFn(
           {
-            hits: extractHitsFromCachedHits(cachedHits),
-            results: undefined,
-            sendEvent,
-            bindEvent,
-            showPrevious,
-            showMore,
-            isFirstPage:
-              helper.state.page === undefined ||
-              getFirstReceivedPage(helper.state, cachedHits) === 0,
-            isLastPage: true,
-            instantSearchInstance,
-            widgetParams,
+            ...this.getWidgetRenderState(initOptions),
+            instantSearchInstance: initOptions.instantSearchInstance,
           },
           true
         );
       },
 
-      render({ results, state, instantSearchInstance }) {
-        const { page = 0 } = state;
-
-        if (escapeHTML && results.hits.length > 0) {
-          results.hits = escapeHits(results.hits);
-        }
-        const initialEscaped = (results.hits as any).__escaped;
-
-        results.hits = addAbsolutePosition(
-          results.hits,
-          results.page,
-          results.hitsPerPage
-        );
-
-        results.hits = addQueryID(results.hits, results.queryID);
+      render(renderOptions) {
+        const { state, instantSearchInstance } = renderOptions;
 
-        results.hits = transformItems(results.hits);
-
-        // Make sure the escaped tag stays after mapping over the hits.
-        // This prevents the hits from being double-escaped if there are multiple
-        // hits widgets mounted on the page.
-        (results.hits as any).__escaped = initialEscaped;
+        const widgetRenderState = this.getWidgetRenderState(renderOptions);
 
         const cachedHits = cache.read({ state }) || {};
```

Comment:
the cache read is still here, but not the cache write, is that expected?

Reply:
maybe we can expose it from getWidgetRenderState as well, something like "currentPage", to use for sendEvent?

It makes sense not to compute the cache read multiple times

Reply:
It is expected, we need it for the `sendEvent` call that must stay in `render`.
One alternative way I thought of was to return `cachedHits` from `getWidgetRenderState` but I don't know if it is better or not.